apps/desktop-wallet/src/features/assetsLists/AddressTokenBadge.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash } from '@alephium/shared'
+
+import useFetchAddressSingleTokenBalances from '@/api/apiDataHooks/address/useFetchAddressSingleTokenBalances'
+import useFetchToken, { isNFT } from '@/api/apiDataHooks/useFetchToken'
+import TokenBadge, { TokenBadgeStyleProps } from '@/components/TokenBadge'
+import { TokenId } from '@/types/tokens'
+
+interface AddressTokenBadgeProps extends TokenBadgeStyleProps {
+  tokenId: TokenId
+  addressHash: AddressHash
+}
+
+const AddressTokenBadge = ({ addressHash, tokenId, ...props }: AddressTokenBadgeProps) => {
+  const { data: token } = useFetchToken(tokenId)
+  const { data, isLoading } = useFetchAddressSingleTokenBalances({ addressHash, tokenId, skip: isNFT(token) })
```

Comment:
Should we also skip if `useFetchToken` is loading?

Reply:
Hm, why would we want to create a request waterfall here? The balance and the token data can run in parallel. We just don't care for NFT balance.

Reply:
Because we need `token` for `isNFT(token)`. Token is not necessarily available, hence I guess that `isNFT(token)` will be false, and `useFetchAddressSingleTokenBalances` will be executed, even if token is an NFT. Or maybe I'm missing something?

Reply:
You are correct. I'll fix it.

Reply:
Fixed in 2d258138d